### ISSUE TITLE
Prefer unstable sort on primitives

### ISSUE THREAD

**Issue by camsteffen** [rust-lang/rust-clippy#5762]
If my assumptions are correct, when sorting a slice of primitive* values, a stable sort is never useful and is less performant.

### What it does

Recommends using `sort_unstable` instead of `sort` when sorting primitive values.

### Categories

- Kind: perf

*What benefit of this lint over old code?*

It's faster and uses less memory.

### Drawbacks

None?

### Example

```rust
let mut vec = vec![2, 1, 3];
vec.sort();
```

Could be written as:

```rust
let mut vec = vec![2, 1, 3];
vec.sort_unstable();
```

*By _primitive_ I mean integers, floating points, bool, char. This could possibly be extended to include arrays, slices and tuples of those types.

**Comment by camsteffen** [comment]
Also, I think it should also include `sort_by` and `sort_by_key`.

**Comment by JarredAllen** [comment]
I can implement this

**Comment by camsteffen** [comment]
@JarredAllen Cool! We're becoming the sort lint team! 😆

**Comment by JarredAllen** [comment]
So I've looked at it more, and I don't think that `sort_by` or `sort_by_key` can be included. It's possible for different values to become equal when using one of those functions, so the unstable sort could give different results than the stable sort (for example, given a vec of integer values, `vec.sort_by_key(|a| a/2)` vs. the unstable equivalent). 

Also, floating-point values can't be sorted because they don't implement `Ord` (because `NaN` compares weirdly), so they have to be dropped from the list of primitive types to detect. Also, even if you get around that, positive and negative zero compare as equal but are distinct, so the stable and unstable sorts are different.

**Comment by JarredAllen** [comment]
I believe this can be closed since my PR was merged (not sure why github didn't do this automatically).

**Comment by ebroto** [comment]
@JarredAllen you must use some specific keywords for that, see [here](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
